package.json

```diff
@@ -36,23 +35,29 @@
     "jsdoc-fresh": "^1.0.2",
     "linkinator": "^1.4.0",
     "mocha": "^6.1.4",
-    "prettier": "^1.17.0"
+    "gts": "^0.9.0",
+    "prettier": "^1.17.0",
+    "typescript": "~3.5.3"
   },
   "scripts": {
     "docs": "jsdoc -c .jsdoc.js",
-    "fix": "eslint . --fix",
-    "lint": "eslint .",
-    "test": "c8 mocha test/*.js",
+    "lint": "gts check && eslint .",
+    "clean": "gts clean",
+    "fix": "gts fix && eslint . --fix",
+    "compile": "tsc -p . && cp -r protos build/",
+    "compile-protos": "compileProtos src",
+    "prepare": "npm run compile-protos && npm run compile",
+    "test": "c8 mocha build/test",
+    "system-test": "c8 mocha --timeout=15000 system-test/*.js",
```

Comment:
We should consider moving our system-tests to TypeScript.

I think it makes sense to leave our sample tests in JavaScript, since not all users are TypeScript users, however, I would also be interested in us starting to provide at least one TypeScript example, perhaps in a folder `typescript-samples`?

Reply:
That's out of scope of the generator. We won't provide system tests, and we also won't provide a `package.json` script for system tests. 

(having said that – I strongly support both ideas, system tests must be converted to TypeScript and we should provide TypeScript samples, but both are not really required to ship TypeScript libraries)